downloads/swan-lake-release-notes/swan-lake-alpha1/RELEASE_NOTE.md

```diff
@@ -402,7 +404,7 @@ The rest descriptor type of the `InclusiveCongifuration` is `anydata` and that o
 
 ##### Improved Listener Declaration  
 
-The listener declaration is improved to accept listener values that are a subtype of both the compiler built-in type `listener` and `error` type. This allows the following listener declaration when the `init` method of `Listener` may return an error.
+The listener declaration is improved to accept listener types that may return an error on initialization. This allows the following listener declaration where the `init` method of `Listener` may return an error.
```

Comment:
@rdhananjaya, please check. 

A value cannot be a subtype of both the listener and error, right? Also the listener declaration doesn't quite accept the value when it is an error. It is only the type that is allowed?

Reply:
+1 for the change, but in `listener T l = expr`, `expr` is a value, not a type.

Reply:
May be we should say `listener class that may return an error ...`

Reply:
Yeah I meant using `T` in the listener declaration when initialization of `T` may return an error.

```ballerina
listener T l1 = new;
listener l2 = new T();
```

+1 for using class. 

How about
```suggestion
The listener declaration is improved to allow using listener classes that may return an error on initialization. This allows the following listener declaration where the `init` method of `Listener` may return an error.
```

Reply:
+1